src/views/name.njk

```diff
@@ -172,17 +173,33 @@
 
 <script>
   document.addEventListener('DOMContentLoaded', function() {
+    function validateNameLength() {
+      const nameLength = document.getElementById('partnership_name').value.length;
+      const suffixLength =  suffixDiv.textContent.length;
+      const maxLength = 160;
+      if (nameLength + suffixLength > maxLength){
+        nameField.setCustomValidity(`Max length including suffix is ${maxLength} characters`);
+      } else {
+        nameField.setCustomValidity('');
+      }
+      nameField.reportValidity();
+    };
+
     const suffixDiv = document.querySelector('.govuk-input__suffix');
     const radioButtons = document.querySelectorAll('input[name="name_ending"]');
 
     radioButtons.forEach(function(radioButton) {
       radioButton.addEventListener('change', function() {
         if (radioButton.checked) {
           suffixDiv.textContent = radioButton.value;
+          validateNameLength();
         }
       });
     });
 
+    const nameField = document.getElementById('partnership_name');
+    nameField.addEventListener('keyup', validateNameLength);
```

Comment:
you could pass nameField into the validateNameLength() function to prevent looking up the name field again. and use nameField.value.length

Reply:
a function used by addEventListener its first parameter is event
it is possible to use it by making event.target and we will get back the field 
but as the function is used without an event listener on line 195 this is not possible